### DISCUSSION TITLE
How to have a child element opt out of a parent's `hx-select`

### DISCUSSION THREAD

**Discussion by angusholder** [bigskysoftware/htmx#681] [Q&A] [answered]
I was hoping I could just write `hx-select=""` on the child to have it opt-out of `hx-select` functionality and use the full result of its `hx-get`, but it seems that `maybeSelectFromResponse` only stops searching when the value of `hx-select` is truthy, rather than non-null, so `hx-select=""` gets ignored entirely.

(Rigjt now I'm working around this by setting `hx-select="#id-of-root-element-of-response"` on the child element, basically turning `hx-select` into a no-op.)

Currently what happens is `getClosestMatch` only stops traversing for when it gets truthy value: https://github.com/bigskysoftware/htmx/blob/6f8899f6fddf0bcf4770eb5387597e9afb578cb5/src/htmx.js#L109-L111

it's called by `getClosestAttributeValue`, which passes `getClosestMatch` the attribute value itself (instead of say a bool): https://github.com/bigskysoftware/htmx/blob/6f8899f6fddf0bcf4770eb5387597e9afb578cb5/src/htmx.js#L119-L125

which is called by `maybeSelectFromResponse`: https://github.com/bigskysoftware/htmx/blob/6f8899f6fddf0bcf4770eb5387597e9afb578cb5/src/htmx.js#L635-L637

I think this could be solved by changing `getClosestAttributeValue` to stop traversing when it reaches a non-null attribute value:
```diff
 function getClosestAttributeValue(elt, attributeName) { 
     var closestAttr = null; 
     getClosestMatch(elt, function (e) { 
-        return closestAttr = getAttributeValue(e, attributeName); 
+        closestAttr = getAttributeValue(e, attributeName); 
+        return closestAttr != null;
     }); 
     return closestAttr; 
 } 
```

Does this sound like a reasonable change/am I going about my issue wrong?

**Comment by 1cg** [comment]
that sounds perfectly reasonable to me, want to put together a pull request?  Are you able to run the test suite?

**Comment by angusholder** [comment]
Sure, I've got the tests running, I'll have a go!

**Comment by 1cg** [comment, marked as answer]
htmx now supports the special `unset` value for all attributes to unset it for all children elements.

See [Attribute Inheritance](https://htmx.org/docs/#inheritance) in the docs.
